frozendict/frozen_dict.py

```diff
@@ -137,11 +137,11 @@ def __deepcopy__(self, memo=None):
 
     def __eq__(self, other):
         if isinstance(other, FrozenDict):
-            return self._dict == other._dict
+            return self.__hash__() == other.__hash__()
```

Comment:
I think hashes should not be used to determine equality, only inequality...

Reply:
why not ? it is the fastest way though...

Reply:
Different objects might have the same hash. Equal object will always have the same hash.
The above is thus not correct. If you want to use the hashes it should rather be:
```
return self.__hash__() == other.__hash__() and self._dict == other._dict
```
But I guess that doing simply `return self._dict == other._dict` will be more efficient in general because this comparison can terminate "eagerly" as soon as some partial inequality is observed, whereas to compare the hashes, they first have to be made, and this process cannot be terminated "eagerly". If one would memoize the hashes on the other hand... but that you pay for with a higher linear memory complexity constant.